server.h

```diff
@@ -188,12 +197,8 @@ class CTFOServer {
           }
           std::shuffle(candidates.begin(), candidates.end(), rng_);
 
-          for (size_t i = 0; i < candidates.size() && (max_count ? (i < max_count) : true); ++i) {
-            const CID cid = candidates[i];
-            response.cards.resize(response.cards.size() + 1);
-            ResponseCardEntry& card_entry = response.cards.back();
-            card_entry.cid = CIDToString(cid);
-            const Card& card = data.Get(cid);
+          auto FillResponseCardEntry = [this](ResponseCardEntry& card_entry, const Card& card) {
```

Comment:
And a nit: Usually, output parameters go last, after the input ones.

(And perhaps some `AddCardToFeedResponse()` would be an easier to digest name then?)